CMake's asciidocor detection is incomplete

### What reproduces the bug?
`cmake ..` succeeds without `asciidoc`, but without it `make` fails. Is it intended behavior?

```
% make
[...]
/bin/sh: 1: ASCIIDOCTOR-NOTFOUND: not found
make[2]: *** [man/adoc/CMakeFiles/adoc_man.dir/build.make:65: man/adoc/bpftrace.8] Error 127
make[2]: *** Deleting file 'man/adoc/bpftrace.8'
make[1]: *** [CMakeFiles/Makefile2:605: man/adoc/CMakeFiles/adoc_man.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/man/adoc/CMakeLists.txt
+++ b/man/adoc/CMakeLists.txt
@@ -1,21 +1,26 @@
 find_program(GZIP gzip REQUIRED)
-find_program(ASCIIDOCTOR asciidoctor REQUIRED)
+find_program(ASCIIDOCTOR asciidoctor)
 file(GLOB FILES *.adoc)
 set(GZFILES "")
-foreach(FIL ${FILES})
-  get_filename_component(NAME ${FIL} NAME_WE)
-  set(MANPAGE_FILE ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.8)
-  set(GZ_MANPAGE_FILE "${MANPAGE_FILE}.gz")
+if(NOT "${ASCIIDOCTOR}" STREQUAL "ASCIIDOCTOR-NOTFOUND")
+  foreach(FIL ${FILES})
+    get_filename_component(NAME ${FIL} NAME_WE)
+    set(MANPAGE_FILE ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.8)
+    set(GZ_MANPAGE_FILE "${MANPAGE_FILE}.gz")
 
-  add_custom_command(OUTPUT ${MANPAGE_FILE}
-    COMMAND ${ASCIIDOCTOR} ${FIL} -b manpage -o - > ${MANPAGE_FILE}
-    DEPENDS ${FIL})
+    add_custom_command(OUTPUT ${MANPAGE_FILE}
+      COMMAND ${ASCIIDOCTOR} ${FIL} -b manpage -o - > ${MANPAGE_FILE}
+      DEPENDS ${FIL})
 
-  add_custom_command(OUTPUT ${GZ_MANPAGE_FILE}
-    COMMAND ${GZIP} -c ${MANPAGE_FILE} > ${GZ_MANPAGE_FILE}
-    DEPENDS ${MANPAGE_FILE})
+    add_custom_command(OUTPUT ${GZ_MANPAGE_FILE}
+      COMMAND ${GZIP} -c ${MANPAGE_FILE} > ${GZ_MANPAGE_FILE}
+      DEPENDS ${MANPAGE_FILE})
 
-  list(APPEND GZFILES ${GZ_MANPAGE_FILE})
-endforeach()
-add_custom_target(adoc_man DEPENDS ${GZFILES})
-install(FILES ${GZFILES} DESTINATION ${CMAKE_INSTALL_MANDIR}/man8)
+    list(APPEND GZFILES ${GZ_MANPAGE_FILE})
+  endforeach()
+  add_custom_target(adoc_man DEPENDS ${GZFILES})
+  install(FILES ${GZFILES} DESTINATION ${CMAKE_INSTALL_MANDIR}/man8)
+else()
+  message(WARNING "asciidoctor not found, building without bpftrace manpage")
+  add_custom_target(adoc_man)
+endif()

